porter/config.go

```diff
@@ -0,0 +1,43 @@
+package porter
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/carolynvs/magex/xplat"
+)
+
+var (
+	// DefaultMixinVersion is the default version of mixins installed when it's not present
+	DefaultMixinVersion = "canary"
+)
+
+// GetPorterHome determines the current PORTER_HOME directory
+func GetPorterHome() string {
+	porterHome := os.Getenv("PORTER_HOME")
+	if porterHome == "" {
+		home, _ := os.UserHomeDir()
+		porterHome = filepath.Join(home, ".porter")
+	}
+	if _, err := os.Stat(porterHome); err != nil {
+		panic("Could not find a Porter installation. Make sure that Porter is installed and set PORTER_HOME if you are using a non-standard installation path")
+	}
+	return porterHome
+}
+
+// UseBinForPorterHome sets the bin/ directory to be PORTER_HOME
+func UseBinForPorterHome() {
+	// use bin as PORTER_HOME
+	wd, _ := os.Getwd()
+	home := filepath.Join(wd, "bin")
+	os.Mkdir(home, 0770)
+	UsePorterHome(home)
+}
+
+// UsePorterHome sets the specified directory to be PORTER_HOME
+func UsePorterHome(home string) {
+	os.Setenv("PORTER_HOME", home)
```

Comment:
Correct, it's not persisted but does last for the duration of the process. I added it, along with EnsurePorter, to the top level mage targets that use porter.

Even if we hard coded our paths to use bin/porter when calling porter, we would still need this env var set to ensure that porter was pulling mixins/plugins from the bin instead of defaulting to ~/.porter.